Code clean ups (inline out var, remove unnecessary assignments)

benaadams opened the pull request:
* Inline out variable declarations

* Remove unnecessary assignments (including using discards for the inlined outs)

benaadams committed:
Inline out variable declarations

benaadams committed:
Remove unnessary assignments

benaadams commented:
Aside VS's `Inline variable declarations` quick option will remove nullable annotations `?` so you have to add them back, not sure where to raise that?

jkotas commented on src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Windows.cs:
```diff
@@ -202,9 +202,8 @@ private static unsafe Interop.Kernel32.SECURITY_ATTRIBUTES GetSecAttrs(FileShare
-            Interop.Kernel32.FILE_STANDARD_INFO info;
 
-            if (!Interop.Kernel32.GetFileInformationByHandleEx(_fileHandle, Interop.Kernel32.FILE_INFO_BY_HANDLE_CLASS.FileStandardInfo, out info, (uint)sizeof(Interop.Kernel32.FILE_STANDARD_INFO)))
+            if (!Interop.Kernel32.GetFileInformationByHandleEx(_fileHandle, Interop.Kernel32.FILE_INFO_BY_HANDLE_CLASS.FileStandardInfo, out Interop.Kernel32.FILE_STANDARD_INFO info, (uint)sizeof(Interop.Kernel32.FILE_STANDARD_INFO)))
```
This makes the line super long.

stephentoub commented on src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs:
```diff
@@ -13,8 +13,7 @@ internal static partial class GlobalizationMode
         {
-            bool exist;
-            bool ret = CLRConfig.GetBoolValue("System.Globalization.Invariant", out exist);
+            bool ret = CLRConfig.GetBoolValue("System.Globalization.Invariant", out bool exist);
```
FWIW, I've generally pushed back on such refactorings.  I know that Visual Studio recommends it, and I know that it reduces the number of lines.  The problem I have is it makes it harder for me to find where things are declared.  It used to be that one only need to look at the method parameters and on the left side of lines to find where things are declared.  With such changes, it now means I may need to effectively look at all of the code in order to find declarations.  I've acquiesced when the out is part of a Try or some such method used in an if condition where the out is only used inside that if block, because then I can naturally just add that one additional place to my mental list of places to look.  I have a harder time swallowing the general case, though.

stephentoub commented:
> Aside VS's Inline variable declarations quick option will remove nullable annotations ? so you have to add them back, not sure where to raise that?

What build are you using?
https://github.com/dotnet/roslyn/issues/38554

benaadams commented:
16.5.0 Preview 1.0
```
Microsoft Visual Studio Enterprise 2019 Preview
Version 16.5.0 Preview 1.0
VisualStudio.16.Preview/16.5.0-pre.1.0+29521.150
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

Visual C++ 2019   00435-60000-00000-AA057
Microsoft Visual C++ 2019

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

AsmDude   1.9.6.14
Syntax highlighting, code completion and code folding for assembly (.asm, .cod, .inc, .s) source code.

ASP.NET and Web Tools 2019   16.5.126.18061
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   16.5.126.18061
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.5.126.18061
Azure App Service Tools v3.0.0

Azure Data Lake Node   1.0
This package contains the Data Lake integration nodes for Server Explorer.

Azure Data Lake Tools for Visual Studio   2.4.1000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   16.5.126.18061
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.4.1000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   3.5.0-beta1-19571-01+fb3f812a3e4b7534bef784fa2df1c21d1f67864d
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

GitHub.VisualStudio   2.10.8.8121
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure HDInsight Azure Node   2.4.1000.0
HDInsight Node under Azure Node

Microsoft Azure Hive Query Language Service   2.4.1000.0
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio   16.0
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service   2.4.1000.0
Language service for Azure Stream Analytics

Microsoft Azure Stream Analytics Node   1.0
Azure Stream Analytics Node under Azure Node

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.21108.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.3+g9c91cdec24
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Node.js Tools   1.5.11023.1 Commit Hash:579896f0984848d17d080c8a1c3ddff98d5ba96f
Adds support for developing and debugging Node.js apps in Visual Studio

NuGet Package Manager   5.5.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Open Command Line   2.4.226
2.4.226

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.61911.21070
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   16.0.11113.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.5.0-beta1-19571-01+fb3f812a3e4b7534bef784fa2df1c21d1f67864d
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual C++ for Linux Development   1.0.9.29521
Visual C++ for Linux Development

Visual F# Tools 10.4 for F# 4.6   16.5.0-beta.19556.3+c2b6d5980a7f64dd4e46448b9b4f5d96eb415aeb
Microsoft Visual F# Tools 10.4 for F# 4.6

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions (Preview)   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio Tools for Kubernetes   1.0
Visual Studio Tools for Kubernetes
```

stephentoub commented:
@benaadams, still working on this?  Want to address the remaining feedback and rebase?

benaadams commented:
> Want to address the remaining feedback and rebase?

I've rebased and addressed @jkotas's comment as he changed the code site in a different PR (hence conflict); so just removed the change.

However, I'm not sure how to address your comment [2004#discussion_r369603091](https://github.com/dotnet/runtime/pull/2004#discussion_r369603091) as it was on one of the simplest `out type ` changes; so should I back out all the `out type ` changes?

stephentoub commented:
> so should I back out all the out type changes?

No, I was mainly ranting. :)

stephentoub commented on src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs:
```diff
@@ -538,7 +538,7 @@ private static void HeapSort(Span<T> keys)
 
             int n = hi - lo + 1;
-            for (int i = n / 2; i >= 1; i = i - 1)
+            for (int i = n / 2; i >= 1; i -= 1)
```
```suggestion
            for (int i = n / 2; i >= 1; i--)
```
?

benaadams replied:
Lol yes, even better

benaadams committed:
Update src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

Co-Authored-By: Stephen Toub <stoub@microsoft.com>

stephentoub commented on src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs:
```diff
@@ -85,8 +85,7 @@ public static string GetFullPath(string path, string basePath)
             int length = path.Length;
-            string? combinedPath = null;
-
+            string? combinedPath;
```
Can this now just be `string combinedPath;`?

benaadams committed:
Update src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs

Co-Authored-By: Stephen Toub <stoub@microsoft.com>

benaadams commented:
Rebased

benaadams commented:
Rebased
